### PR TITLE
[6-57]Creation guards to prevent access to the different roles

### PR DESCRIPTION
**Pull request by HenadzV**
1.  Greated guards for HR and Coach poles

### CODE REVIEW

File: exalearn-ui/src/app/guards/role-coach.guard.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	providedIn: 'root'
+})
+export class RoleCoachGuard implements CanActivate {
+	constructor(private userService: UserService, private router: Router) {}
+
+	canActivate(
+		route: ActivatedRouteSnapshot,
+		state: RouterStateSnapshot
+	): Observable<boolean> | Promise<boolean> | boolean {
+		if (`${this.userService.currentUser?.role.toLocaleLowerCase()}` !== 'coach') {
```
Comment (vstrabko) [inline exalearn-ui/src/app/guards/role-coach.guard.ts]: Don't we have a enum with all user roles defined?
If we don't it makes sense to create one and make sure we use it everywhere.

**Review by vstrabko — CHANGES_REQUESTED** [review]

File: exalearn-ui/src/app/guards/role-hr.guard.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	providedIn: 'root'
+})
+export class RoleHRGuard implements CanActivate {
+	constructor(private userService: UserService, private router: Router) {}
+
+	canActivate(
+		route: ActivatedRouteSnapshot,
+		state: RouterStateSnapshot
+	): Observable<boolean> | Promise<boolean> | boolean {
+		if (`${this.userService.currentUser?.role.toLocaleLowerCase()}` !== 'hr') {
```
Comment (vstrabko) [inline exalearn-ui/src/app/guards/role-hr.guard.ts]: Same here.

**Review by vstrabko — CHANGES_REQUESTED** [review]

**Comment by vstrabko** [comment]
Other than that - looks nice.
Happy to approve.

File: exalearn-ui/src/app/guards/role-coach.guard.ts (hunk 1 above)
Comment (HenadzV) [reply exalearn-ui/src/app/guards/role-coach.guard.ts]: Done, also used new enum in RoleService

File: exalearn-ui/src/app/guards/role-hr.guard.ts (hunk 2 above)
Comment (HenadzV) [reply exalearn-ui/src/app/guards/role-hr.guard.ts]: Same)

**Review by GeranSkot — APPROVED** [review]

**Review by HrebenYury — APPROVED** [review]

**Review by KotdenGit — APPROVED** [review]

**Review by boburwebdev — APPROVED** [review]

**Review by vstrabko — APPROVED** [review]

**Pull request merged**